### PR TITLE
fix: Fix multiple spawn of SpeechInputlistner

### LINKED ISSUE #1713: App crash due to multiple spawning of voice input
**Actual Behaviour**

The fragment which is called when voice input is pressed does not disable the voice input and thus if the user presses on the same location more than once multiple voice input is activated and if it's more than twice then the app crashes

**Expected Behaviour**

The voice input should be disabled for the entire duration until the input is given to susi

**Steps to reproduce it**
Press on the voice input button multiple times (even if it's not visible)

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by kartik-karz**
Fixes #1713

Changes: Voice input is disabled until the previous input is running

### CODE REVIEW

**Comment by kartik-karz** [comment]
https://github.com/fossasia/susi_android/issues/1713

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt [hunk 1]
```diff
@@ -285,6 +285,12 @@ class ChatActivity : AppCompatActivity(), IChatView {
             }
         }
 
+
+    }
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: What is being closed here?

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt [hunk 2]
```diff
@@ -341,6 +345,7 @@ class ChatActivity : AppCompatActivity(), IChatView {
             chatPresenter.check(true)
             btnSpeak.setImageResource(R.drawable.ic_mic_24dp)
             btnSpeak.setOnClickListener({
+                btnSpeak.isClickable = false
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: Should we disable the button instead?

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt [hunk 3]
```diff
@@ -122,6 +122,9 @@ class STTfragment : Fragment() {
 
     override fun onPause() {
         super.onPause()
+        if ( activity is ChatActivity) {
+            (activity as ChatActivity).enableVoiceInput()
+        }
         (activity as ChatActivity).fabsetting.show()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: Move this inside the if block as well and remove the cast.

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt [hunk 4]
```diff
@@ -122,6 +122,9 @@ class STTfragment : Fragment() {
 
     override fun onPause() {
         super.onPause()
+        if ( activity is ChatActivity) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: Extra space after `(`

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt (hunk 3 above)
Comment (kartik-karz) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: move the fabsetting.show()? shouldn't I do it in another PR? wouldn't it confuse others if they look at it later?

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt (hunk 2 above)
Comment (kartik-karz) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: both are kinda similar but disabled seems more intuitive, will change that.

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt (hunk 3 above)
Comment (kartik-karz) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: also @iamareebjamal I think the cast is required by the compiler, it's an error if we remove the special cast.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: I thought it should work through Kotlin smart cast. Let me check and get back to you

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt (hunk 1 above)
Comment (kartik-karz) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: the code was written between the last brace and so it felt as though there was an extra brace, it's resolved in the latest commit

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt [hunk 5]
```diff
@@ -122,8 +122,11 @@ class STTfragment : Fragment() {
 
     override fun onPause() {
         super.onPause()
-        (activity as ChatActivity).fabsetting.show()
+	    if (activity is ChatActivity) {
+            (activity as ChatActivity).enableVoiceInput()
+            (activity as ChatActivity).fabsetting.show()
```
[the remark this answers is not in the document]
Comment (kartik-karz) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: > I thought it should work through Kotlin smart cast. Let me check and get back to you

In our previous chat here https://github.com/fossasia/susi_android/pull/1714#discussion_r221815328
I explained the problem and you mentioned that you would test it out, was this after the test?

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: Yes it will work

[the remark this answers is not in the document]
Comment (kartik-karz) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: @iamareebjamal 
2018-10-12-200013_1366x768_scrot
I don't know why but I seem to have error compiling it right from that day, are you using some special option to make ide neglect that error or are you testing it directly using gradlew?

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: I checked general smart cast. It is not possible with activities, please continue with older fix

**Pull request merged**